File: intentresolver/src/main/java/com/walhalla/intentresolver/TiktokIntent.java
package com.walhalla.intentresolver;

import static com.walhalla.intentresolver.IntentUtils.resolveMp4ActivitiesForPackage;

import android.content.Context;

import com.walhalla.abcsharedlib.SharedNetwork;

import java.io.File;

public class TiktokIntent extends BaseIntent{

    public TiktokIntent() {
        super(SharedNetwork.Package.TIKTOK_M_PACKAGE);
    }

    @Override
    public void shareMp4Selector(Context context, File file) {
        resolveMp4ActivitiesForPackage(context, file, packageName);
    }

    @Override
    public void videoShare(Context context, String path) {

    }
}

File: intentresolver/src/main/java/com/walhalla/intentresolver/TiktokIntent.kt
package com.walhalla.intentresolver

import android.content.Context
import com.walhalla.abcsharedlib.SharedNetwork
import java.io.File

class TiktokIntent : BaseIntent(SharedNetwork.Package.TIKTOK_M_PACKAGE) {

    override fun shareMp4Selector(context: Context, file: File) {
        IntentUtils.resolveMp4ActivitiesForPackage(context, file, packageName)
    }


    override fun videoShare(context: Context, path: String) {
    }
}
